packages/hyperion-util/src/guid.ts

```diff
@@ -8,5 +8,5 @@
 export type GUID = string;
 
 export function guid(): GUID {
-  return 'f' + (Math.random() * (1 << 30)).toString(16).replace('.', '');
+  return 'f' + (window.crypto.getRandomValues(new Uint32Array(1))[0]).toString(16).replace('.', '');
```

Comment:
Thanks for updating this. This code can run outside of main thread (e.g. in WebWorker). To be safe you should remove the `window` and instead use `globalThis`. It is also safer to use feature detection and first check to make sure `globalThis.crypto` is available.

e.g.
```
const getRandom : ()=>number = globalThis.cryto 
  ? ()=> globalThis.crypto.getRandomValues(new Uint32Array(1))[0]
  : ()=> Math.random()
;
export function guid(): GUID {
  return 'f' + (getRandom()).toString(16).replace('.', '');
}
```